### PR TITLE
Changed RadioButtonGroup to support numeric values

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed RadioButtonGroup to support numeric values.

This allows setting something to `5` instead of `"5"`.

#### Where should the reviewer start?

doc.js

#### What testing has been done on this PR?

added unit test

#### How should this be manually tested?

see unit test

#### Do the grommet docs need to be updated?

automatic

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/RadioButtonGroup/index.d.ts [hunk 1]
```diff
@@ -5,8 +5,8 @@ export interface RadioButtonGroupProps {
   disabled?: boolean;
   name: string;
   onChange?: ((event: React.ChangeEvent<HTMLInputElement>) => void);
-  options: (string | { disabled?: boolean, id?: string, label?: (string | React.ReactNode), value: string})[];
-  value?: string | object;
+  options: (string | { disabled?: boolean, id?: string, label?: (string | React.ReactNode), value: string | number })[];
+  value?: string | number | object;
```
Comment (ShimiSun) [inline src/js/components/RadioButtonGroup/index.d.ts:9]: Should we remove the `object` option?

Comment (ericsoderberghp) [reply src/js/components/RadioButtonGroup/index.d.ts:9]: nope, we still allow options to contain objects. But, I reviewed the typescript types again and found another issue, fixed.

Comment (ShimiSun) [reply src/js/components/RadioButtonGroup/index.d.ts:9]: In this case, should it be added to PropTypes in doc.js?

Comment (ericsoderberghp) [reply src/js/components/RadioButtonGroup/index.d.ts:9]: update to add value as number to prop types

**Pull request merged**